### PR TITLE
docs: add all remaining controller configurations

### PR DESCRIPTION
**Pull request by agilgur5**
### Motivation

Several configurations were missing, and, as a result, entirely undocumented
  - several of these can be very useful for different use-cases, lack of docs makes these settings hard to discover
  - addresses https://github.com/argoproj/argo-workflows/pull/11293#issuecomment-1622162284 and https://github.com/argoproj/argo-workflows/issues/8539

#### Primary Changes
- add namespace, columns, podSpecLogStrategy, podGCGracePeriodSeconds, podGCDeleteDelayDuration, initialDelay, and retentionPolicy
  - add comments for all of the above, primarily copying from `config.go`: https://github.com/argoproj/argo-workflows/blob/30a916880ff21027ad1004c4ae176037f1a956da/config/config.go#L22
    - a few comments from other places, like `info.go` for Column: https://github.com/argoproj/argo-workflows/blob/8ec5623093181cfb5ef7c34a5df8a8c917282b7b/pkg/apis/workflow/v1alpha1/info.go#L18
    - `argo-helm` for retentionPolicy: https://github.com/argoproj/argo-helm/blob/5f55ef2c4cc16d4c91faaf1bdb7e0c1d7385435a/charts/argo-workflows/values.yaml#L322
      - big kudos to the `argo-helm` maintainers and contributors for keeping all the config up-to-date despite lack of docs!

#### Minor Secondary Changes
- fix typing issues -- configmap values must be strings
- minor copy-edits to docs and comments
  - add comments for RetentionPolicy as it was the odd one out with no comments
  - fix comment that said DeleteDelayDuration was integer seconds when it is actually a Duration type

### Verification

n/a

### CODE REVIEW

**Comment by agilgur5** [comment]
Might need to run codegen on this. On a different computer, so gotta set up my dev environment again 😅

**Review by terrytangyuan — COMMENTED** [review]
@jmeridth Would you like to help review this PR? This borrows some docs from argo-helm :-)

**Comment by agilgur5** [comment]
added a tiny fix to existing docs. realized the comment for DeleteDelayDuration incorrectly stated the type as `int` when it is a `Duration`

**Comment by agilgur5** [comment]
@jmeridth just a friendly follow-up that this is awaiting your review 🙂

File: docs/workflow-controller-configmap.yaml [hunk 1]
```diff
@@ -78,6 +81,16 @@ data:
       scope: workflow-list
       url: http://workflows?label=workflows.argoproj.io/completed=true
 
+  # Columns are custom columns that will be exposed in the Workflow List View.
+  columns: |
```
Comment (jmeridth) [inline docs/workflow-controller-configmap.yaml:85]: this is follow-up to [this PR](https://github.com/argoproj/argo-workflows/pull/10808), correct?

**Review by jmeridth — APPROVED** [review]
based on my understanding this looks good

File: docs/workflow-controller-configmap.yaml (hunk 1 above)
Comment (agilgur5) [reply docs/workflow-controller-configmap.yaml:85]: Yep! Docs/example for that feature

**Review by terrytangyuan — APPROVED** [review]

File: docs/workflow-controller-configmap.yaml (hunk 1 above)
Comment (agilgur5) [reply docs/workflow-controller-configmap.yaml:85]: added a comment that this apparently requires Argo v3.5 (I had not realized that earlier) in #11474

**Pull request merged**